### PR TITLE
Bugfix/aqm ee2

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

The following EE2 edits are made

- Sets COMOUT correctly
- Adds KEEPDATA and SENDMAIL to driver and jobs file
- Moves J-jobs under jobs
- Removes unneeded comments

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Run the prep, stats, and plots drivers.

Remember for stats, that you need to set -v cyc=04 or some other cycle.
For plots, set COMIN to emc.vpppg

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

ASAP

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

File: dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh [hunk 1]
```diff
@@ -45,43 +45,19 @@ export mod_ver=${aqm_ver}
...
-## setting to produce output to personal directory
-## export COMIN=/lfs/h2/emc/physics/noscrub/$USER/${NET}/${evs_ver}
-## export COMIN=/lfs/h2/emc/vpppg/noscrub/$USER/${NET}/${evs_ver}
 export COMIN=/lfs/h2/emc/vpppg/noscrub/$USER/${NET}/${evs_ver}
 export COMINaqm=/lfs/h2/emc/vpppg/noscrub/$USER/evs/v1.0/stats/aqm/aqm
 
-export COMOUT=/lfs/h2/emc/ptmp/$USER/$NET/$evs_ver/${STEP}/${COMPONENT}
+export COMOUT=/lfs/h2/emc/ptmp/$USER/$NET/$evs_ver
 #
 ## export KEEPDATA=NO
```
Comment (AliciaBentley-NOAA) [inline dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh]: Can we remove this commented out line from the dev driver?

**Comment by AliciaBentley-NOAA** [comment]
I have reviewed this PR and ensured that it checks all the AQM boxes on the EE2 checklist. Thanks!

File: dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh (hunk 1 above)
Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh]: Done.

File: jobs/JEVS_AQM_PLOTS [hunk 2]
```diff
@@ -59,13 +61,11 @@ export VDATE=${VDATE:-${PDYm3}}
 
 export COMIN=${COMIN:-$(compath.py ${envir}/com/${NET}/${evs_ver}/)}
 export COMINaqm=${COMINaqm:-$(compath.py ${envir}/com/${NET}/${evs_ver}/stats/${COMPONENT}/${MODELNAME})}
-export COMOUT=${COMOUT:-${COMIN}/${STEP}/${COMPONENT}}
-export COMOUTplots=${COMOUTplots:-${COMOUT}/${RUN}.${VDATE}}
-export COMOUTheadline=${COMOUTheadline:-${COMOUT}/headline.${VDATE}}
+export COMOUT=${COMOUT:-$(compath.py -o ${envir}/com/${NET}/${evs_ver}/)}
```
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_AQM_PLOTS]: When compath.py is used w/ -o, do not use `${envir}/com`. Please update this definition to be: `export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver}/)}`

File: modulefiles/aqm/aqm_plots.sh
Comment (ShelleyMelchior-NOAA) [inline modulefiles/aqm/aqm_plots.sh:1]: Please correct to `$prod_util_ver`

File: modulefiles/aqm/aqm_prep.sh
Comment (ShelleyMelchior-NOAA) [inline modulefiles/aqm/aqm_prep.sh:1]: Please correct to `$prod_util_ver`

File: modulefiles/aqm/aqm_stats.sh
Comment (ShelleyMelchior-NOAA) [inline modulefiles/aqm/aqm_stats.sh:1]: Please correct to `$prod_util_ver`

**Comment by ShelleyMelchior-NOAA** [comment]
I ran aqm prep processing. Output looks good and matches emc.vpppg parallel.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/aqm/atmos.20230929/aqm/`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_prep_00.o83352782`

File: dev/drivers/scripts/aqm/stats/jevs_aqm_grid2obs_stats.sh [hunk 3]
```diff
@@ -45,18 +45,16 @@ source $config
...
 export job=${PBS_JOBNAME:-jevs_${MODELNAME}_${VERIF_CASE}_${STEP}}
 export jobid=$job.${PBS_JOBID:-$$}
 
 export cycle=t${cyc}z
-export VDATE=$(date --date="3 days ago" +%Y%m%d)
 
-export COMOUT=/lfs/h2/emc/vpppg/noscrub/$USER/${NET}/${evs_ver}/${STEP}/${COMPONENT}
-export COMOUTsmall=${COMOUT}/${RUN}.${VDATE}/${MODELNAME}/${VERIF_CASE}
-export COMOUTfinal=${COMOUT}/${MODELNAME}.${VDATE}
+export COMIN=/lfs/h2/emc/vpppg/noscrub/$USER/${NET}/${evs_ver}
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/aqm/stats/jevs_aqm_grid2obs_stats.sh:54]: COMIN is defined twice: lines 46 and 56. Let's keep one. I vote line 56.

File: scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh [hunk 4]
```diff
@@ -64,8 +61,14 @@ obs_hourly_found=0
 if [ -s ${check_file} ]; then
     obs_hourly_found=1
 else
-    echo "Can not find pre-processed obs hourly input ${check_file}"
-    ## add email function here
+  echo "Can not find pre-processed obs hourly input ${check_file}"
+  if [ $SENDMAIL = "YES" ]; then 
+    export subject=AQM Hourly Observed Missing for EVS ${COMPONENT}"
```
Comment (ShelleyMelchior-NOAA) [inline scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh]: Add an opening ". The line should be: `export subject="AQM Hourly Observed Missing for EVS ${COMPONENT}"`

File: scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh [hunk 5]
```diff
@@ -221,7 +228,14 @@ if [ -s ${check_file} ]; then
     obs_daily_found=1
 else
     echo "Can not find pre-processed obs daily input ${check_file}"
-    ## add email function here
+    if [ $SENDMAIL = "YES" ]; then
+      export subject=AQM Daily Observed Missing for EVS ${COMPONENT}"
```
Comment (ShelleyMelchior-NOAA) [inline scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh]: Add an opening ". The line should be: `export subject="AQM Hourly Observed Missing for EVS ${COMPONENT}"`

**Comment by ShelleyMelchior-NOAA** [comment]
I ran aqm stats processing but it did not produce stats. I am trying to track down the problem.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o83353373`

Aha! A typo in the definition of the subject of your emails for the SENDMAIL test. I made the fixes mentioned above in this PR conversation in my local testing and the stats are all produced and compare well with emc.vpppg parallel.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/aqm/atmos.20230929/aqm/grid2obs`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o83354560`

**Comment by ShelleyMelchior-NOAA** [comment]
aqm plots has been submitted and completed. 
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots.o83355042`
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/aqm/atmos.20230930`
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/aqm/headline.20230930`
The tar files compare well in size against emc.vpppg.

File: scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh (hunk 4 above)
Comment (PerryShafran-NOAA) [reply scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh]: I wonder if lack of this " is what caused the errant email to be sent to Ho-Chun.

Comment (ShelleyMelchior-NOAA) [reply scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh]: @PerryShafran-NOAA , perhaps! It made the log file all wonky. Like the log trace did not match the script at all. I was finally able to track it down in the code itself. Like, you know how the code can change color or highlight open and closed brackets, etc ...? The coloring was funny in the SENDMAIL test block and I traced it to a missing ".

File: scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh (hunk 5 above)
Comment (PerryShafran-NOAA) [reply scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh]: Fixed.

File: scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh (hunk 4 above)
Comment (PerryShafran-NOAA) [reply scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh]: Oh, no, it was probably the typo you had mentioned below.

Comment (PerryShafran-NOAA) [reply scripts/aqm/stats/exevs_aqm_grid2obs_stats.sh]: This error is fixed.

File: dev/drivers/scripts/aqm/stats/jevs_aqm_grid2obs_stats.sh (hunk 3 above)
Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/aqm/stats/jevs_aqm_grid2obs_stats.sh:54]: Fixed.

File: modulefiles/aqm/aqm_stats.sh
Comment (PerryShafran-NOAA) [reply modulefiles/aqm/aqm_stats.sh:1]: Fixed.

File: modulefiles/aqm/aqm_prep.sh
Comment (PerryShafran-NOAA) [reply modulefiles/aqm/aqm_prep.sh:1]: Fixed.

File: modulefiles/aqm/aqm_plots.sh
Comment (PerryShafran-NOAA) [reply modulefiles/aqm/aqm_plots.sh:1]: Fixed.

File: jobs/JEVS_AQM_PLOTS (hunk 2 above)
Comment (PerryShafran-NOAA) [reply jobs/JEVS_AQM_PLOTS]: Fixed.

**Comment by ShelleyMelchior-NOAA** [comment]
I think this PR is resolved and ready for merge, but for a sanity check, I am re-running stats to make sure no more errant emails are delivered.

**Comment by ShelleyMelchior-NOAA** [comment]
The re-run of aqm stat has completed. VDATE=20230929 for cyc=04
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/aqm/atmos.20230929/aqm/grid2obs`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o83397637`
If @Ho-ChunHuang-NOAA and @AliciaBentley-NOAA didn't get an email from me, then I think we are good to go w/ this PR.

**Comment by PerryShafran-NOAA** [comment]
There's no indication in the .o file that any mail was sent (I would assume that if they were, it would be indicated in the .o file).  

Stats look good.

I think we're good to go now.

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi,

I did not receive any email notification for this run.

Thanks.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
All tests performed as expected. This PR is ready for merge.

**Pull request merged**